client/dashboard/app/queries/site-activity-log.ts

```diff
@@ -1,9 +1,19 @@
 import { queryOptions } from '@tanstack/react-query';
-import { fetchSiteActivityLog } from '../../data/site-activity-log';
+import { fetchSiteActivityLog, fetchSiteRewindableActivityLog } from '../../data/site-activity-log';
 
 export const siteLastFiveActivityLogEntriesQuery = ( siteId: number ) =>
 	queryOptions( {
 		queryKey: [ 'site', siteId, 'activity-log', 'last-five' ],
 		queryFn: () => fetchSiteActivityLog( siteId, { number: 5 } ),
 		select: ( data ) => data.current?.orderedItems?.slice( 0, 5 ) ?? [],
 	} );
+
+export const siteRewindableActivityLogEntriesQuery = (
+	siteId: number,
+	number: number = 1000 // 1000 is the maximum number of entries that can be fetched.
```

Comment:
Are we saying that the endpoint can only fetch 1000 data in a single page, or that we only store the last 1000 entries, or that we only want to support showing the latest 1000 entries in the dashboard?

Asking because we might want to switch to backend pagination (instead of using client-side `filterSortAndPaginate()`) if there's too many data from the backend. But we can save it for later.